design-proposal.md

```diff
@@ -0,0 +1,66 @@
+# Proposal: Design of the rkt + Kubernetes CRI
+
+## Abstract
+
+This proposal can be viewed as more a discussion than an actual proposal, because we still haven't reached a consensus on the architecture.
+
+## Backgroud
+
+As container runtime interface (CRI) proposals and original stub codes are already merged into Kubernetes, the actual work has already begun.
+All existing and future container runtimes are expected to conform the interface and provides feature parities.
+
+## Design(s)
+
+There are three different designs:
+
+- `Kubelet` --_gPRC calls_--> `rktlet service` --_invoke 'rkt' binary, or use some rkt library_--> result
```

Comment:
I've thought that it probably makes sense to isolate the `datadir` for system-rkt vs kubelet-rkt. If we default to that and/or recommend it then that can resolve the second concern.

I think the first concern is really work that is already needed for `api-service`, so unless we get rid of that too the 1st concern doesn't matter all that much.